src/algebra/lie/of_associative.lean

```diff
@@ -62,6 +62,8 @@ instance of_associative_ring : lie_ring A :=
 
 lemma of_associative_ring_bracket (x y : A) : ⁅x, y⁆ = x*y - y*x := rfl
 
+lemma lie_apply {α : Type*} (f g : α → A) (a : α) : ⁅f, g⁆ a = ⁅f a, g a⁆ := rfl
```

Comment:
This lemma belongs much earlier I expect, right when we define a lie structure on pi types

Reply:
I think is fine where it is. The bracket here is coming from `ring.has_bracket`, just 20 lines higher.

The ring structure on `(α → A)` comes from `pi.ring` which is distant but I think that as long as `ring.has_bracket` is here then this lemma belongs here too.

Reply:
Oh, I was assuming this was a proof about a `pi.lie_ring` structure, which would apply more generally. I guess that doesn't exist yet, and is out of scope for this PR.

Reply:
Oh, yes it should.